### PR TITLE
remove kepler maps and spatial-data-catalog in on-premise

### PR DESCRIPTION
**Pull request by Josmorsot**
Remove sections, Kepler and Spatial data catalog based on `cartodb_com_hosted`

https://app.clubhouse.io/cartoteam/story/158492/remove-kepler-maps

### CODE REVIEW

**Review by rafatower — APPROVED** [review]
:+1: from my limited Vue knowledge

the only thing: I'd add an entry to the `NEWS.md` and let all the tests run (including building the images). Otherwise it won't let you merge

image

**Comment by VictorVelarde** [comment]
@Josmorsot If we want to remove kepler.gl maps we should also consider the 'Recent content' list at Home page: --> https://github.com/CartoDB/cartodb/blob/e37b2f49c7259540c943fe23d251f48b4e67665c/lib/assets/javascripts/new-dashboard/pages/Home/RecentSection/RecentSection.vue#L66

eg: 

image

**Comment by Jesus89** [comment]
Should we also skip the routes of the spatial-data-catalog if isOnprem?

**Comment by Jesus89** [comment]
Routes for the catalog: https://github.com/CartoDB/cartodb/blob/ed7a7731670574ec2f0b6cf0aa1fa21619c254c6/lib/assets/javascripts/new-dashboard/router/routes/data.js#L110L140

These actions should be hidden for onprem: https://github.com/CartoDB/cartodb/blob/ed7a7731670574ec2f0b6cf0aa1fa21619c254c6/lib/assets/javascripts/new-dashboard/components/QuickActions/DatasetQuickActions.vue#L73L82

This is another link should be hidden for onprem: https://github.com/CartoDB/cartodb/blob/ed7a7731670574ec2f0b6cf0aa1fa21619c254c6/lib/assets/javascripts/builder/dataset/dataset-content/dataset-content-options.tpl#L6 used here https://github.com/CartoDB/cartodb/blob/ed7a7731670574ec2f0b6cf0aa1fa21619c254c6/lib/assets/javascripts/builder/dataset/dataset-content/dataset-content-options-view.js#L45.

I guess the subscriptions are only shown if DO is enabled, so nothing to change there (https://github.com/CartoDB/cartodb/tree/ed7a7731670574ec2f0b6cf0aa1fa21619c254c6/lib/assets/javascripts/new-dashboard/components/Subscriptions) I think.

**Comment by rafatower** [comment]
Please take a look at this https://app.clubhouse.io/cartoteam/story/143678/make-do-catalog-visible-from-dashboard and lets discuss with @ernesmb , as there is some clashing requirement in there with regards to DO and catalog

**Comment by Jesus89** [comment]
Yep, I saw the issue. I'm just adding more internal links to the catalog. But @Josmorsot pointed out that these appear only for subscriptions, and the subscriptions page only if DO is enabled, so we are OK.

I would only hide the routes then

**Comment by Josmorsot** [comment]
All changes has been commited. 

About  https://app.clubhouse.io/cartoteam/story/143678/make-do-catalog-visible-from-dashboard >>
I'm waiting for @ernesmb because if I understand now we want to load catalog from `public.carto.com/api...`  when frontend is running in an on-premise.

So if we want that, then we should undo some of the last commits

**Comment by Josmorsot** [comment]
I've just talked with  @ernesmb:
https://app.clubhouse.io/cartoteam/story/143678/make-do-catalog-visible-from-dashboard it's going to be closed. Spatial data catalog is going to be removed from on-premises, so we can go with this PR.

**Comment by rafatower** [comment]
Adding @jgoizueta for awareness and coordination

**Review by Jesus89 — APPROVED** [review]
Code LGTM.

This will require also deep testing in an on-prem release

File: lib/assets/javascripts/new-dashboard/pages/Home/RecentSection/RecentSection.vue [hunk 1]
```diff
@@ -63,7 +63,13 @@ export default {
       isFetching: state => {
         return state.recentContent.isFetching;
       },
-      recentContent: state => state.recentContent.list
+      recentContent: state => {
+        if (state.config.cartodb_com_hosted) {
+          return state.recentContent.list.filter(l => l.type !== 'keplergl');
```
Comment (VictorVelarde) [inline lib/assets/javascripts/new-dashboard/pages/Home/RecentSection/RecentSection.vue:68]: 👍🏻

**Review by VictorVelarde — APPROVED** [review]
👍🏻

**Comment by rafatower** [comment]
Related to [ch143678]

File: lib/assets/javascripts/new-dashboard/components/NavigationBar/NavigationBar.vue [hunk 2]
```diff
@@ -45,13 +45,13 @@
           :message="$t('FeedbackMessage.message')"
           @click.native.stop.prevent="toggleDropdown"/>
 
-        <NotificationPopup
-          v-if="!popupWasShown('popups.DataObservatorySamples')"
+        <!-- <NotificationPopup
```
Comment (rafatower) [inline lib/assets/javascripts/new-dashboard/components/NavigationBar/NavigationBar.vue:48]: So I understand this is gonna be disabled to everyone, right?

Comment (rafatower) [reply lib/assets/javascripts/new-dashboard/components/NavigationBar/NavigationBar.vue:48]: related to [ch158573] if I'm not mistaken

**Comment by rafatower** [comment]
Related to [ch158573]

File: lib/assets/javascripts/new-dashboard/components/NavigationBar/NavigationBar.vue (hunk 2 above)
Comment (Josmorsot) [reply lib/assets/javascripts/new-dashboard/components/NavigationBar/NavigationBar.vue:48]: Yes, we've removed for everyone in all environments

**Pull request merged**